src/sr25519.rs

```diff
@@ -786,6 +786,73 @@ pub unsafe extern "C" fn sr25519_relay_vrf_modulo_assignments_cert(
     }
 }
 
+/// Makes verification with the given VRF proof and output and calculates tranche value
+/// @param public_key_ptr byte representation of the public key that was used to sign the message
+/// @param output_ptr the signature
+/// @param proof_ptr the proof of the signature
+/// @param n_delay_tranches - the number of delay tranches in total.
+/// @param zeroth_delay_tranche_width - the zeroth delay tranche width.
+/// @param relay_vrf_story - relay vrf story
+/// @param core_index - leaving core for computing
+/// @param tranche_out - output tranche
+#[allow(unused_attributes)]
+#[no_mangle]
+pub unsafe extern "C" fn sr25519_vrf_verify_and_get_tranche(
+    public_key_ptr: *const u8,
+    output_ptr: *const u8,
+    proof_ptr: *const u8,
+    n_delay_tranches: u32,
+    zeroth_delay_tranche_width: u32,
+    relay_vrf_story: *const RelayVRFStory,
+    core_index: u32,
+    tranche_out: *mut u32,
+) -> Sr25519SignatureResult {
+    assert!(!public_key_ptr.is_null());
+    assert!(!output_ptr.is_null());
+    assert!(!proof_ptr.is_null());
+    assert!(!relay_vrf_story.is_null());
+    assert!(!tranche_out.is_null());
+
+    let public_key = create_public(slice::from_raw_parts(
+        public_key_ptr,
+        SR25519_PUBLIC_SIZE as usize,
+    ));
+
+    let tranche_out = std::mem::transmute::<*mut u32, &mut u32>(tranche_out);
+
+    let relay_vrf_story =
+        std::mem::transmute::<*const RelayVRFStory, &RelayVRFStory>(relay_vrf_story);
+
+    let vrf_output = match VRFOutput::from_bytes(slice::from_raw_parts(
+        output_ptr,
+        SR25519_VRF_OUTPUT_SIZE as usize,
+    )) {
+        Ok(val) => val,
+        Err(err) => return convert_error(&err),
+    };
+
+    let vrf_proof = match VRFProof::from_bytes(slice::from_raw_parts(
+        proof_ptr,
+        SR25519_VRF_PROOF_SIZE as usize,
+    )) {
+        Ok(val) => val,
+        Err(err) => return convert_error(&err),
+    };
+
+    let (vrf_in_out, _) = match public_key.vrf_verify(
+        relay_vrf_delay_transcript(relay_vrf_story.clone(), core_index),
+        &vrf_output,
+        &vrf_proof,
+    ) {
```

Comment:
Hm, wouldn't something like
```suggestion
    let (vrf_in_out, _) = public_key.vrf_verify(
        relay_vrf_delay_transcript(relay_vrf_story.clone(), core_index),
        &vrf_output,
        &vrf_proof,
    ).map_err(convert_error)?;
```
work? This match pattern gets very repetitive here, it's better to either wrap it in a macro or do something like the suggestion above.